slave_state.go

```diff
@@ -81,15 +61,42 @@ func newSlaveStateCollector(httpClient *httpClient, userTaskLabelList []string)
 						for _, label := range t.Labels {
 							normalisedLabel := normaliseLabel(label.Key)
 							// Ignore labels not explicitly whitelisted by user
-							if inArray(normalisedLabel, normalisedUserTaskLabelList) {
+							if stringinSlice(normalisedLabel, normalisedUserTaskLabelList) {
 								taskLabels[normalisedLabel] = label.Value
 							}
 						}
 						c.(*prometheus.GaugeVec).With(taskLabels).Set(1)
 					}
 				}
 			}
-		},
+		}
+	}
+
+	if len(slaveAttributeLabelList) > 0 {
+		normalisedAttributeLabels := normaliseLabelList(slaveAttributeLabelList)
+		AttributeLabelList := append(defaultLabels, normalisedAttributeLabels...)
+		metrics[prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Help:      "Slave attributes",
+			Namespace: "mesos",
+			Subsystem: "slave",
+			Name:      "attributes",
+		}, AttributeLabelList)] = func(st *slaveState, c prometheus.Collector) {
+			slaveAttributesExport := map[string]string{
+				"slave": st.PID,
```

Comment:
Same as above: A the master has its on `attributes` metric containing the `slave` label, I don't think we need to repeat this here.